### PR TITLE
doc: document call argument evaluation order

### PR DESCRIPTION
**Pull request by alandonovan**
..now that the spec is resolved.

### CODE REVIEW

File: doc/spec.md [hunk 1]
```diff
@@ -990,6 +990,12 @@ same name, such as `f(x=1, x=2)`. A call that provides a `**kwargs`
 argument may yet have two values for the same name, such as
 `f(x=1, **dict(x=2))`. This results in a dynamic error.
 
+Function arguments are evaluated in the order they appear in the call.
+
+Unlike Python, Starlark does not allow more than `*args` argument in a
```
Comment (jayconrod) [inline doc/spec.md]: Should this be "does not allow more than one"?

File: internal/compile/compile.go [hunk 2]
```diff
@@ -1567,27 +1567,23 @@ func (fcomp *fcomp) args(call *syntax.CallExpr) (op Opcode, arg uint32) {
...
+	// evaluation order. This is both easy to implement and
+	// consistent with lexical order:
 	//
-	// For now we implement the Python2 semantics, but
-	// the spec needs to clarify the correct approach.
-	// Perhaps it would be best if we statically rejected
-	// named arguments after *args (e.g. y=5) so that the
-	// Python2 implementation strategy matches lexical order.
-	// Discussion in github.com/bazelbuild/starlark#13.
+	//   f(1, 2, x=3, *[4], **dict(z=6)) # 1 2 3 5 6
```
Comment (jayconrod) [inline internal/compile/compile.go]: `# 1 2 3 4 6`

File: doc/spec.md (hunk 1 above)
Comment (alandonovan) [reply doc/spec.md]: Um, yeah.

File: internal/compile/compile.go (hunk 2 above)
Comment (alandonovan) [reply internal/compile/compile.go]: It's a good thing one of us is awake today.

**Comment by alandonovan** [comment]
PTAL

**Pull request merged**